Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm

```diff
@@ -412,6 +415,35 @@ String certDataBase64(""
     return components;
 }
 
+#if HAVE(SEC_KEYCHAIN)
+static SecKeychainRef getTempKeychain()
+{
+    SecKeychainRef keychainRef = NULL;
+    uuid_t uu;
+    char pass[PATH_MAX];
+
+    uuid_generate_random(uu);
+    uuid_unparse_upper(uu, pass);
+    UInt32 passLen = (UInt32)strlen(pass);
+    NSString* path = [NSString stringWithFormat:@"%@/%s", NSTemporaryDirectory(), pass];
+
+    [NSFileManager.defaultManager removeItemAtPath:path error:NULL];
+    RELEASE_ASSERT(SecKeychainCreate(path.fileSystemRepresentation, passLen, pass, false, NULL, &keychainRef) == errSecSuccess);
+    RELEASE_ASSERT(keychainRef);
+    RELEASE_ASSERT(SecKeychainUnlock(keychainRef, passLen, pass, TRUE) == errSecSuccess);
```

Comment:
Do we need a corresponding SecKeychainLock call?  Also, RELEASE_ASSERT is probably a bit excessive here.  EXPECT_TRUE, EXPECT_NOT_NULL, or EXPECT_EQ cause a test failure without a crash.

Reply:
I don't think I want to lock this keychain, then I'd just need to immediately unlock it. This is just a throw-away keychain for API which needs a keychain to operate correctly.

I will change the macros.